### PR TITLE
Extend SIGs

### PR DESCRIPTION
**Pull request by fabiand**
**What this PR does / why we need it**:
#8858 proposed to introduce SMEs, SIGs, *and* reassign ownership - all at once.
This might be too much, thus this PR starts by sanitizing group names and adding the `sig-` prefix and adding empty SME groups.

In the next step members can be assigned and code can be moved to create tangible areas (SIGs for pkgs, SMEs for knowledge areas) of ownership.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by fabiand** [comment]
cc @xpivarc

File: OWNERS_ALIASES [hunk 1]
```diff
@@ -44,16 +44,24 @@ aliases:
       - alicefr
       - 0xFelix
       - lyarwood
-  test-reviewers:
+
```
Comment (rthallisey) [inline OWNERS_ALIASES:47]: How about a sig-api reviewers?  Kubernetes has something similar where a group approves API changes based on design, feature quality, api conventions, ect... 

Also I could see a sig-compute.   The group would handle anything that influences how a vm gets scheduled, e.g. devices, numa, virt-launcher mem/cpu footprint, ect..

Comment (xpivarc) [reply OWNERS_ALIASES:47]: +1

File: OWNERS_ALIASES [hunk 2]
```diff
@@ -87,3 +87,13 @@ aliases:
...
   sig-storage-reviewers: {}
+
+  #
+  # SMEs
+  #
+  sme-live-migration: {}
+  sme-api-and-controllers: {}
+  sme-scheduling-and-node-plumbing: {}
+  sme-scale-and-observability: {}
+  sme-ci-and-test-framework: {}
```
Comment (xpivarc) [inline OWNERS_ALIASES:98]: We have a sig-test already. I think I would just introduce the ci, maybe even as sig.

Comment (fabiand) [reply OWNERS_ALIASES:98]: Yes, we have sig-test already.
However, we need to reflect that it's two groups: people who own and create tests (usually sigs actually), and people who provide the framework and infra to run them.

Agree that it's not well defined, and something we should clean up, therefore I'm removing this line for now.

File: OWNERS_ALIASES (hunk 1 above)
Comment (fabiand) [reply OWNERS_ALIASES:48]: Agreed, thanks !

File: OWNERS_ALIASES [hunk 3]
```diff
@@ -63,9 +71,29 @@ aliases:
...
+  sig-storage-approvers: {}
+  sig-storage-reviewers: {}
+
+  #
+  # SMEs
+  #
+  sme-live-migration: {}
+  sme-api-and-controllers: {}
+  sme-scheduling-and-node-plumbing: {}
+  sme-scale-and-observability: {}
```
Comment (fabiand) [inline OWNERS_ALIASES]: Dropping scale here, as it's simply too redundant with sig-scale

**Comment by rmohr** [comment]
/lgtm

I like introducing the empty groups to lay out the structure, and then rework code positions to fill it over time, like in #9648.

That looks like a good managable path forward. Thanks @fabiand for being persistent on pushing.

**Comment by vasiliy-ul** [comment]
This looks good! I have Just a small question. Maybe it has been discussed somewhere already (and sorry if that is the case) but why now we drop the `sme-*` naming in favor of `sig-*`? 

```
OWNER_ALIAS: Split SMEs and SIGify 

Split SMEs into smaller scopes, and align to the SIG naming
```

I see that https://github.com/kubevirt/kubevirt/pull/8858 introduces both `sme-*` and `sig-*` names. While here we have only `sig-*` and the PR description still says that we distinguish between those : `SIGs for pkgs, SMEs for knowledge areas`. Won't it cause confusion?

**Comment by fabiand** [comment]
@vasiliy-ul hey - it's a long journey by now, and indeed this change (dropping SMEs) was not further discussed.

My reasoning in this update: I did not have the impression that the idea of SIGs being "vertical" and SMEs connecting them "horizontally" ("weaving a net") was well understood. Because of this, and for simplicity I aligned on the term SIGs.
Because the scope of these (now) SIGs did not change, the fact that some SIGs are horizontal and some are vertical has not change. It#s just not reflected in the prefix anymore.

I'm good to bring SMEs back if you think that it was helpful.

File: OWNERS_ALIASES [hunk 4]
```diff
@@ -64,9 +73,83 @@ aliases:
...
+  # SIG Storage
+  # Owns anything related to storage.
+  #
+  sig-storage-approvers: {}
+  sig-storage-reviewers: {}
+
+  #
+  # SIG API
+  # Owns the API including ensuring API life-cycle, including
+  # deprecation, and backwards compatibility.
```
Comment (vasiliy-ul) [inline OWNERS_ALIASES:98]:
```suggestion
  # Owns the API including API life-cycle, deprecation
  # and backwards compatibility.
```

**Review by vasiliy-ul — APPROVED** [review]
> I'm good to bring SMEs back if you think that it was helpful.

No, I am fine with having only SIGs. Thanks for clarifying. 

Added just one nit inline.

LGTM

**Comment by fabiand** [comment]
@xpivarc please review

**Comment by xpivarc** [comment]
@fabiand All good, tide will try to merge this in next batch. I will adjust my PR to this.

**Comment by smg247** [comment]
@fabiand, I believe this is causing an error in the `periodic-autoowners` job: 
```
time="2023-05-01T07:39:12Z" level=info msg="handling repo ..." orgRepo=kubevirt/kubevirt
time="2023-05-01T07:39:12Z" level=info msg="GetFile(kubevirt, kubevirt, OWNERS, )" client=github
time="2023-05-01T07:39:12Z" level=info msg="GetFile(kubevirt, kubevirt, OWNERS_ALIASES, )" client=github
time="2023-05-01T07:39:12Z" level=error msg="Unable to parse aliases config." error="error unmarshaling JSON: while decoding JSON: json: cannot unmarshal object into Go struct field .aliases of type []string"
```
I believe that it cannot handle the empty list notation you have utilized.
[logs](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-prow-auto-owners/1652938582622474240)

**Comment by dhiller** [comment]
> @fabiand, I believe this is causing an error in the `periodic-autoowners` job:
> 
> ```
> time="2023-05-01T07:39:12Z" level=info msg="handling repo ..." orgRepo=kubevirt/kubevirt
> time="2023-05-01T07:39:12Z" level=info msg="GetFile(kubevirt, kubevirt, OWNERS, )" client=github
> time="2023-05-01T07:39:12Z" level=info msg="GetFile(kubevirt, kubevirt, OWNERS_ALIASES, )" client=github
> time="2023-05-01T07:39:12Z" level=error msg="Unable to parse aliases config." error="error unmarshaling JSON: while decoding JSON: json: cannot unmarshal object into Go struct field .aliases of type []string"
> ```
> 
> I believe that it cannot handle the empty list notation you have utilized. [logs](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-prow-auto-owners/1652938582622474240)

This is strange, I thought the verify-owners plugin should have caught this.

**Comment by dhiller** [comment]
See #9677

**Comment by dhiller** [comment]
@smg247 ^^

**Pull request merged**